packages/app/src/debug/DebugContainer.cy.tsx

```diff
@@ -281,37 +283,151 @@ describe('<DebugContainer />', () => {
+      cy.findByTestId('debug-spec-item').should('be.visible')
+    })
+
+    it('simulates full running run with failed tests', () => {
+      //Change this value when modifying this test to allow time to see transitions
+      const waitTimeBetweenSimulatedEvents = 0
+
+      cy.mountFragment(DebugSpecsFragmentDoc, {
+        variableTypes: DebugSpecVariableTypes,
+        variables: {
+          hasNextRun: false,
+          runNumber: 1,
+          nextRunNumber: -1,
+        },
+        onResult: (result) => {
+          if (result.currentProject?.cloudProject?.__typename === 'CloudProject') {
+            const test = result.currentProject.cloudProject.runByNumber
+
+            //creating copy to prevent mutation later on in this test
+            const cloudRunCopy = JSON.parse(JSON.stringify(CloudRunStubs.failingWithTests))
+
+            result.currentProject.cloudProject.runByNumber = {
+              ...cloudRunCopy,
+              status: 'RUNNING',
               runNumber: 1,
-              completedInstanceCount: 5,
-              totalInstanceCount: 6,
+              completedInstanceCount: 2,
+              totalInstanceCount: 3,
             } as typeof test
           }
         },
-        render: (gqlVal) => <DebugContainer gql={gqlVal} />,
+        render: (gqlVal) => {
+          return <DebugContainer gql={gqlVal} />
+        },
+      }).then(({ component }) => {
+        cy.wrap(component.gql).as('gql')
       })
 
+      /*
+        DebugTestingProgress is not mocked here resulting in it always
+        saying "0 of 0 specs completed"
+        Trying to mock up the DebugTestingProgress here results in
+        Urql clearing out and refetching the main query which makes
+        this test fail.
+      */
+
+      const getRun = (gql: DebugSpecsFragment) => {
+        const run = gql.currentProject?.cloudProject?.__typename === 'CloudProject'
+        && gql.currentProject.cloudProject.runByNumber
+
+        if (!run) {
+          throw Error('Could not find run')
+        }
+
+        return run
+      }
+
       cy.findByTestId('debug-header').should('be.visible')
-      cy.findByTestId('debug-pending-splash')
-      .should('be.visible')
+      cy.findByTestId('debug-testing-progress').should('be.visible')
+
+      cy.wait(waitTimeBetweenSimulatedEvents)
+      cy.get<DebugSpecsFragment>('@gql').then((gql) => {
+        const run = getRun(gql)
+
+        cy.wrap(run).as('run')
+
+        run.totalFailed = 4
+        const newTest = JSON.parse(JSON.stringify(run.testsForReview[0]))
+
+        newTest.id = '345'
+        newTest.thumbprint = `${newTest.thumbprint}c`
+
+        run.testsForReview.push(newTest)
+      })
+
+      cy.wait(waitTimeBetweenSimulatedEvents)
+      cy.findAllByTestId('debug-spec-item').should('have.length', 1)
+      cy.get<CloudRun>('@run').then((run) => {
+        const newSpec: CloudSpecRun = JSON.parse(JSON.stringify(run.specs[0]))
```

Comment:
This is fine - no change request - but just out of interested, I found out about `window.structuredClone` recently, which is similar to this parse/stringify trick: https://developer.mozilla.org/en-US/docs/Web/API/structuredClone

Neat little function!

Reply:
I had not seen that before.  I will check it out.  I just needed something basic to make sure I could mutate the test data without affecting other tests.